examples/webhook-signing/nextjs/api-router/route.ts

```diff
@@ -0,0 +1,43 @@
+import { NextRequest, NextResponse } from 'next/server';
+import Stripe from 'stripe';
+import { headers } from 'next/headers'
+import { getErrorMessage, logError } from '@/lib/error-methods';
+
+// Initialize Stripe
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY!);
+
+// Webhook handler
+export async function POST(req: NextRequest) {
+  const webhookSecret: string = process.env.STRIPE_WEBHOOK_SECRET!;
+  const headersList = headers()
```

Comment:
`headers` is an async function [ref](https://nextjs.org/docs/app/api-reference/functions/headers), and we need to use `await` here . For example
```
const headersList = await headers();
```

Alternatively, we can get the headers directly from `req`
```
const headersList = req.headers;

```